craft_parts/sources/git_source.py

```diff
@@ -220,22 +219,23 @@ def _clone_new(self):
             command.extend(["--branch", self.source_tag or self.source_branch])
         if self.source_depth:
             command.extend(["--depth", str(self.source_depth)])
-        self._run(command + [self.source, self.part_src_dir])
+        self._run(command + [self.source, str(self.part_src_dir)])
 
         if self.source_commit:
             self._fetch_origin_commit()
             command = [
                 self.command,
                 "-C",
-                self.part_src_dir,
+                str(self.part_src_dir),
```

Comment:
you should not need to str things passed to subprocess, right?

Reply:
Yes, subprocess should do the right thing when pathlib.Path is used.